File: java-workshop/part-1-powermock/src/main/java/repository/OrderRepository.java
package repository;

import domain.Order;
import domain.OrderStatus;
import infrastructure.Database;

public class OrderRepository {

    public Order findById(String orderId) {
        return Database.getOrder(orderId);
    }

    public void updateStatus(String orderId, OrderStatus status) {
        Database.updateOrderStatus(orderId, status);
    }

    public void save(Order order) {
        Database.saveOrder(order);
    }
}

File: csharp-workshop/Part1.StaticDeps/Repository/OrderRepository.cs
using Part1.StaticDeps.Domain;
using Part1.StaticDeps.Infrastructure;

namespace Part1.StaticDeps.Repository;

public class OrderRepository
{
    public Order? FindById(string orderId) =>
        Database.GetOrder(orderId);

    public void UpdateStatus(string orderId, OrderStatus status) =>
        Database.UpdateOrderStatus(orderId, status);

    public void Save(Order order) =>
        Database.SaveOrder(order);
}
